morango/sync/operations.py

```diff
@@ -968,6 +968,18 @@ def handle(self, context):
         if records_transferred > 0:
             _dequeue_into_store(context.transfer_session)
 
+        # update database max counters but use latest fsics from client/server
+        fsic = (
+            context.transfer_session.client_fsic
+            if context.is_server
+            else context.transfer_session.server_fsic
+        )
+        DatabaseMaxCounter.update_fsics(
+            json.loads(fsic),
+            context.filter,
+            v2_format=FSIC_V2_FORMAT in context.capabilities,
+        )
+
```

Comment:
We'll want to ensure this happens within the same transaction as the dequeuing itself, since otherwise the database could still end up in an inconsistent state. I think moving this inside `_dequeue_into_store` would be sufficient, given that function is wrapped with `@transaction.atomic(using=USING_DB)`.

Reply:
(we also don't need to worry about needing this to be run if no records were transferred)